.github/workflows/website.yml

```diff
@@ -48,30 +68,77 @@ jobs:
       - uses: bahmutov/npm-install@v1
         env:
           PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: 1
-      - run: npm run build -w packages/website
+      - run: touch .env
+      - run: |
+          (
+            npm -w '@web3-storage/website' run ci:envfile --silent \
+            | sed "s/^NEXT_PUBLIC_MAGIC=$/NEXT_PUBLIC_MAGIC=$NEXT_PUBLIC_MAGIC/" \
+            | sed "s/^MAGIC_SECRET_KEY=$/MAGIC_SECRET_KEY=$MAGIC_SECRET_KEY/" \
+            ;
+            echo 'NEXT_PUBLIC_MAGIC_TESTMODE_ENABLED="1"';
+          ) > .env
```

Comment:
Can you give me some more context on what's going on here? In general, we should be able to just set the ENV in CI, rather than being forced to write things to a .env file... those things are supposed to be a local dev convenience, rather than a sed inducing beast.

Reply:
> Can you give me some more context on what's going on here?

def

> In general, we should be able to just set the ENV in CI, rather than being forced to write things to a .env file

oh yeah. I totally I tried that... like a lot. I wanted to avoid writing the MAGIC_SECRET_KEY to the fs and just have it in the env...

This is the only thing I could get to work. It has to do with `dotenv`... which I rarely use so perhaps I'm missing something. 

The other thing is how api uses miniflare here. https://github.com/web3-storage/web3.storage/blob/main/packages/api/package.json#L30
* See how miniflare uses `--env ../../.env`. In my testing, it will *only* use that file, and will not inherit other env vars. The only way I could get the API to have these env vars set at runtime was to write them into the `.env` file, and it has to do with how the api package starts with `miniflare` cli

With all that in mind... is there another way you'd want me to do it? The only other thing I could think of involved meddling with the `miniflare` npm script in api package, e.g. using the 'binding' flag. but this seemed less gross

Whether or not we make a bigger change in how to provide these values dynamically like this, I will add a comment to this part of the workflow file

Reply:
btw I could also get rid of the use of `sed` by putting that find/replace logic in `npm -w '@web3-storage/website' run ci:envfile` and having that script read from envs or its own flags, which seems like a nice way of encapsulating this in a node script with no `sed`

Reply:
@gobengo that's interesting... in https://github.com/nftstorage/nft.storage/pull/2130 I'm overriding some env variables in the current nodejs process and running the `miniflare` cli using `execa`, and miniflare does seem to use the overrides instead of what's in the .env file.

But I've also seen what you're describing with the current setup - running `yarn dev` only grabs the vars from the file... Now I'm curious what the difference is...

Reply:
> With all that in mind... is there another way you'd want me to do it?

Let's figure out how we can set env vars for miniflare in a way that doesn't force us to not use actual ENV vars. I'm hopeful we can fix that without changing miniflare itself.

Reply:
@olizilla I found a way: https://github.com/web3-storage/web3.storage/pull/1781